### PR TITLE
Add Maintenance and Hub services

### PR DESCRIPTION
**Pull request by LilianaLessa**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| **Issue**         | https://mundipagg.atlassian.net/browse/MOD-560
| **What?**         | Add Maintenance and Hub services
| **Why?**          | To improve the maintainability of the module. Hub service is a requirement to get the right keys.

### CODE REVIEW

File: src/Kernel/Aggregates/Charge.php [hunk 1]
```diff
@@ -292,9 +292,7 @@ public function getLastTransaction()
         $newest = $transactions[0];
 
         foreach ($transactions as $transaction) {
-            if (
-                $newest->getCreatedAt()->getTimestamp() <
-                $transaction->getCreatedAt()->getTimestamp()
+            if ($newest->getCreatedAt()->getTimestamp() <$transaction->getCreatedAt()->getTimestamp()
```
Comment (michelpl) [inline src/Kernel/Aggregates/Charge.php]: Corrigir

File: src/Maintenance/Services/InfoBuilderService.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+{
+    /**
+     *
+     * @param  array $query
+     * @return string|array
+     */
+    public function buildInfoFromQueryArray(array $query)
+    {
+        $infos = [];
+        if ($this->isTokenValid($query)) {
```
Comment (michelpl) [inline src/Maintenance/Services/InfoBuilderService.php]: Desaninhar ifs

File: src/Maintenance/Services/InstallDataSource/CoreInstallDataSource.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Mundipagg\Core\Maintenance\Services\InstallDataSource;
+
```
Comment (GabrielDeveloper) [inline src/Maintenance/Services/InstallDataSource/CoreInstallDataSource.php:4]: linha extra

File: src/Maintenance/Services/InstallDataSource/CoreInstallDataSource.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+{
+    private $integrityFilePath;
+
+    public function __construct()
+    {
+        $dir = explode(DIRECTORY_SEPARATOR, __DIR__);
+        array_pop($dir);
+        array_pop($dir);
+        $dir = implode(DIRECTORY_SEPARATOR, $dir);
+
```
Comment (GabrielDeveloper) [inline src/Maintenance/Services/InstallDataSource/CoreInstallDataSource.php:15]: linha extra

File: src/Maintenance/Services/IntegrityInfoRetrieverService.php [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+        foreach ($dirCount as $part) {
+            $part = array_keys($part);
+            $part = end($part);
+            $rootDir .= $part . DIRECTORY_SEPARATOR;
+        }
+
+        return $rootDir;
+    }
+
+    private function generateIntegrityFile($dataSource)
```
Comment (GabrielDeveloper) [inline src/Maintenance/Services/IntegrityInfoRetrieverService.php]: adicionat typehint

**Review by GabrielDeveloper — APPROVED** [review]

File: src/Maintenance/Services/InstallDataSource/AbstractInstallDataSource.php [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+                foreach ($foundFiles as $foundFile) {
+                    if (strlen($foundFile) < 3 
+                        || (                        $ignoreVendor 
+                        && strpos($foundFile, 'vendor') !== false)
+                    ) {
+                        continue;
+                    }
+
+                    $foundFile = $dir . DIRECTORY_SEPARATOR . $foundFile;
+                    $foundFile = preg_replace('/\\' .DIRECTORY_SEPARATOR. '{2,}/', DIRECTORY_SEPARATOR, $foundFile);
```
Comment (michelpl) [inline src/Maintenance/Services/InstallDataSource/AbstractInstallDataSource.php]: Quebrar linha

File: src/Maintenance/Services/IntegrityInfoRetrieverService.php [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+            'removed' => count($removed),
+            'added' => count($added),
+            'files' => $processedFiles,
+            'reference' => count($integrityData),
+        ];
+        $integrityInfo->files = $fileHashs;
+        $integrityInfo->reference = $integrityData;
+
+        return $integrityInfo;
+
```
Comment (michelpl) [inline src/Maintenance/Services/IntegrityInfoRetrieverService.php]: Remover linha extra

File: src/Maintenance/Services/LogDownloadInfoRetrieverService.php [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+        $params = explode(':', $value);
+
+        try {
+            $extension = $params[0];
+            $file = base64_decode($params[1]);
+        } catch (\Throwable $e) {
+            return null;
+        }
+
```
Comment (michelpl) [inline src/Maintenance/Services/LogDownloadInfoRetrieverService.php:24]: Remover linha extra

**Review by michelpl — APPROVED** [review]

**Pull request merged**